File: park-address.php
<?php
	// Get cURL resource
	$curl = curl_init();
	// Set options
	$park = 'acad';
	$dataURL = 'https://developer.nps.gov/api/v0/parks?parkCode=' . $park . "&fields=addresses";
	curl_setopt_array($curl, array(
		CURLOPT_RETURNTRANSFER => true,
		CURLOPT_URL => $dataURL,
		CURLOPT_USERAGENT => $_SERVER['HTTP_USER_AGENT'],
		CURLOPT_HTTPHEADER => array('Authorization: INSERT-API-KEY-HERE')
	));
	// Send the request & save response to $response
	$response = curl_exec($curl);
	// Close request to clear up some resources
	curl_close($curl);
	$json = json_decode($response);
	
	// Prepare and execute address output
	echo $json->data[0]->fullName, "<br />";
	for ($i = 0; $i < count($json->data[0]->addresses); $i++) {
		if ($json->data[0]->addresses[$i]->type == "Mailing") {
			echo $json->data[0]->addresses[$i]->line1, "<br />";
			if ($json->data[0]->addresses[$i]->line2 != "") {
				echo $json->data[0]->addresses[$i]->line2, "<br />";
			}
			if ($json->data[0]->addresses[$i]->line3 != "") {
				echo $json->data[0]->addresses[$i]->line3, "<br />";
			}
			echo $json->data[0]->addresses[$i]->city, ", ", $json->data[0]->addresses[$i]->stateCode, " ", $json->data[0]->addresses[$i]->postalCode;
		}
	}
?>

File: park-address.py
import urllib.request, json

# Configure API request
park = "acad"
endpoint = "https://developer.nps.gov/api/v0/parks?parkCode=" + park + "&fields=addresses"
HEADERS = {"Authorization":"INSERT-API-KEY-HERE"}
req = urllib.request.Request(endpoint,headers=HEADERS)

# Execute request and parse response
response = urllib.request.urlopen(req).read()
data = json.loads(response.decode('utf-8'))

# Prepare and execute output
print(data["data"][0]["fullName"])
for address in data["data"][0]["addresses"]:
    if address["type"] == "Mailing":
        print(address["line1"])
        if address["line2"] != "":
            print(address["line2"])
        if address["line3"] != "":
            print(address["line3"])
        print(address["city"] + ", " + address["stateCode"] + " " + str(address["postalCode"]))
